components/atoms/ProductCount/ProductCount.tsx

```diff
@@ -7,15 +10,27 @@ const ProductCount: React.FC<ProductCountProps> = ({
   count,
   className,
   ...props
-}) => (
-  <div
-    {...props}
-    className={['w-full text-2xs uppercase text-primary', className ?? ''].join(
-      ' ',
-    )}>
-    {/* TODO: i8n */}
-    All Products ({count})
-  </div>
-);
+}) => {
+  const lang = useSettingsStore((state) => state.settings?.lang);
+
+  const productCountLabel = fallbackString(
+    lang?.categories?.productCount,
+    'All products ({count})',
+  );
+  const parsedProductCountLabel = parseTextWithVariables(productCountLabel, {
+    count: count.toString(),
+  });
```

Comment:
We should have one place only to parse these `lang` objects & configs, we already have quite a bunch in utils/lang.ts.

We can also simplify the approach here, something like `i18n('categories.product_count', {count})` should be more desirable and easier to work with.